Fix(?) [and clean up] Android CI configuration

hawkrives opened the pull request:
Last night's Android nightly didn't complete successfully; https://circleci.com/gh/StoDevX/AAO-React-Native/49056.

```
Exit status of command '/home/circleci/project/android/gradlew assembleRelease -p ./android' was 1 instead of 0.

FAILURE: Build failed with an exception.

* Where:
Build file '/home/circleci/project/android/app/build.gradle' line: 195

* What went wrong:
A problem occurred evaluating project ':app'.
> Could not set unknown property 'hasFile' for project ':app' of type org.gradle.api.Project.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 0s
```

In #3557, I did the following, because _my_ gradle wouldn't build the debug version because it couldn't find `bundleReleaseJsAndAssets`.

```diff
gradle.projectsEvaluated {
    // hook bundleData into the android build process
+    if (project.hasProperty('bundleDebugJsAndAssets')) {
        bundleDebugJsAndAssets.dependsOn bundleData
+    }
+    if (project.hasProperty('bundleReleaseJsAndAssets')) {
        bundleReleaseJsAndAssets.dependsOn bundleData
+    }
}
```

My hypothesis here is that we're somehow running an old version of Gradle on Circle, since we don't track the `android/gradle/wrapper/gradle-wrapper.properties` file (which controls the version) in our cache key.

So!

I did ~~two~~ three things here:

1. I adjusted the `load-*` cache instructions to use the same key as the `save-*` cache instructions, to prevent accidental typos between the two
2. I split the cache keys with spaces instead of hyphens, and I split them across multiple lines in the source, to make it easier to read and to add new items (see the relevant bit of YAML syntax: [playground](https://nodeca.github.io/js-yaml/#yaml=LSAmZm9vID4tCiAgdjEKICBncmFkbGUKICBkZXBlbmRlbmNpZXMKICB7eyBhcmNoIH19CiAge3sgY2hlY2tzdW0gImFuZHJvaWQvYnVpbGQuZ3JhZGxlIiB9fQogIHt7IGNoZWNrc3VtICJhbmRyb2lkL3NldHRpbmdzLmdyYWRsZSIgfX0KICB7eyBjaGVja3N1bSAiYW5kcm9pZC9hcHAvYnVpbGQuZ3JhZGxlIiB9fQogIHt7IGNoZWNrc3VtICJub2RlX21vZHVsZXMvcmVhY3QtbmF0aXZlL3BhY2thZ2UuanNvbiIgfX0K), [SO answer](https://stackoverflow.com/a/21699210))
3. I added `{{ checksum "android/gradle/wrapper/gradle-wrapper.properties" }}` and `{{ checksum "android/settings.gradle" }}` to the cache key. 
    1. `gradle-wrapper.properties` tracks the version of Gradle that we want to use
    2. `android/settings.gradle` is YAADF (yet another android dependencies file), so I just added it for completeness

hawkrives committed:
remove duplicative listing of cache keys

hawkrives committed:
split cache key definitions over several lines

hawkrives committed:
add gradle version file to the cache key

hawkrives committed:
add additional android dep config file to the cache key

drewvolz reviewed (approved):
RSLGTM. I'll defer to @rye.

rye commented on .circleci/config.yml:
```diff
@@ -35,14 +38,21 @@ commands:
     steps:
       - save_cache:
-          key: 'v1-gradle-dependencies-{{ arch }}-{{ checksum "android/build.gradle" }}-{{ checksum "android/app/build.gradle" }}-{{ checksum "node_modules/react-native/package.json" }}'
+          key: &gradle-key >-
```
I'll assume that since Circle didn't barf when it got this that it will actually handle it correctly. Please change to `gradle-cache-key`.

rye commented on .circleci/config.yml:
```diff
@@ -13,14 +13,17 @@ commands:
     steps:
       - save_cache:
-          key: 'v2-ruby-dependencies-{{ arch }}-{{ checksum "Gemfile.lock" }}'
+          key: &bundler-key >-
```
Please change to `bundler-cache-key`, as this signifies _exactly_ what it's for.

rye commented on .circleci/config.yml:
```diff
@@ -66,14 +76,17 @@ commands:
     steps:
       - save_cache:
-          key: 'v3-yarn-dependencies-{{ arch }}-{{ checksum "yarn.lock" }}'
+          key: &yarn-key >-
```
`yarn-cache-key` please

rye committed:
circle: Replace \w*-key with \w*-cache-key in context

I prefer it when the YAML anchors are named to what they actually
represent.

  $ circleci config validate
  Config file at .circleci/config.yml is valid.

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>
Tested-by: Kristofer Rye <kristofer.rye@gmail.com>
